### DISCUSSION TITLE
does Halfedge need endVert?

### DISCUSSION THREAD

**Discussion by wtholliday** [elalish/manifold#959] [General]
In my own Halfedge code, I don't have an `endVert` because `h.endVert == halfedges[h.next].startVert`

Asking because I'd like to reduce memory usage.

thanks!

**Comment by pca006132** [comment]
see https://github.com/elalish/manifold/pull/939#issuecomment-2359059551

**Reply by pca006132** [reply]
If we really want to remove it from halfedge, we can, but some code may become more complicated. If you want to remove it, you can try and show us the memory usage and timing differences, and we can accept it if the improvement is significant.

**Reply by elalish** [reply]
There are some reasons I think that'll be a pain. And honestly, if you're looking at memory usage, our halfedges are a pretty small fraction anyway.

**Reply by pca006132** [reply]
Agreed, we haven't looked into optimizing memory usage til now. We spent most of our time around correctness, getting some features done, and improving the runtime.

@wtholliday If you are interested, you can use tracy memory profiling to look at which part is using the most memory. You need to enable some cmake flags but iirc those are documented in the readme.

**Reply by wtholliday** [reply]
Ok, yeah I made assumptions about memory usage based on my own half-edges being kinda significant (~100mb) and Manifold's Halfedge being double the size of mine. I'll try to profile and see what's up :)

**Reply by pca006132** [reply]
halfedge size is not insignificant when you have large meshes, just that we have things that are more memory consuming than that... which are the sparse indices in boolean3.

It may be possible to reduce the size, see https://github.com/elalish/manifold/pull/931, but it is non-trivial to do.
